tests/common/system_utils/docker.py

```diff
@@ -99,7 +99,6 @@ def download_image(dut, registry, image_name, image_version="latest"):
             image_version (str): The version of the image to download.
     """
 
-    dut.command("docker login {} -u {} -p {}".format(registry.host, registry.username, registry.password))
```

Comment:
We still need this part if we're able to find `registry_vars` on line 54. To make this work, I think you can just check if username and password are present. If they are, then do the `login` first. If they are not, then skip the login and use the anonymous method. :)

Reply:
Also you can delete the check from line 73-76 as we are now OK if username and password are not found in the hostvars.

Reply:
Deleted 2 checkers there.

---
In reply to: [467336245](https://github.com/Azure/sonic-mgmt/pull/2069#discussion_r467336245) [](ancestors = 467336245)

Reply:
Resolved

---
In reply to: [467336136](https://github.com/Azure/sonic-mgmt/pull/2069#discussion_r467336136) [](ancestors = 467336136)